authz/authz.proto

```diff
@@ -192,6 +217,8 @@ message ProbeRequest {
 // ProbeResponse returns the ACK/NACK for a single user request
 // as evaluated against the current policy, along with the version of the policy
 // that the gRPC call/user were evaluated against.
+// If no policy has been defined, the default response is ACTION_PERMIT, with
+// a zero-length version string.
```

Comment:
If no policy has been defined, a similar thing has to be defined even for GetResponse right?

For both Probe() and Get(), would it make sense to return FAILED_PRECONDITION error which signifies that the Rotate() has not been performed? Because for both Probe() and Get(), Rotate() is a precondition.

Reply:
Is Rotate() a precondition?  Why wouldn't (or shouldn't) Prove or Get work without beginning with a Rotate()?

I do think that you are correct about FAILED_PRECONDITION if no policy has been set yet.

Reply:
What should Get() return when there is no policy?

```
message GetResponse {
    string version = 1;     -> Should it be an empty string?
    uint64 created_on = 2;  -> Should it be 0?
    string policy = 3;      -> Should it be an empty string?
}
```

Reply:
I think you misunderstood; I agree with you that if no policy has been defined, Get() should return the error.

But, if no Rotate() has been issued to start a stream, Get() and Probe() should still work, while Get() would still fail if no policy has been set previously.

Reply:
> I do think

Sorry I misread it as "I don't think". My bad

Oh yeah. Get() and Probe() does not have a direct dependency on Rotate() stream.
Yes, I was referring to when the policy is not yet loaded, what should be the Get's behaviour. And I guess we both agree on Get() returning error.

Shouldn't Probe() also return the same error? Just my thought to keep Probe() and Get() consistent if no policy has been set previously.
Will wait for other's opinion

Reply:
IMO, Probe() should return what will actually happen if a RPC is executed; so, if a policy has not been defined, it will always return "allowed", rather than being a special case.

Reply:
going three times, resolved.